src/js/stores/search.js

```diff
@@ -0,0 +1,258 @@
+         * Reset or initialize store state.
+         *
+         * @private
+         */
+        _handleReset: function () {
+            this._registeredSearches = {};
+            this._registeredSearchTypes = {};
+        },
+
+        /**
+         * Update the search items for the specified search module
+         *
+         * @private
+         * @param {string} id The search module ID to update
+         */
+        _update: function (id) {
+            this._updateSearchItems(id);
+        },
+
+        /**
+         * Get the search items for the specified search module
+         *
+         * @param {string} id The search module ID 
+         * @return {Immutable.Iterable.<object>}
+         */
+        getSearchItems: function (id) {
+            return this._registeredSearches[id].searchItems;
+        },
+
+        /**
+         * Get the filters for the specified search module
+         *
+         * @param {string} id The search module ID 
+         * @return {Array.<Array.<string>>}
+         */
+        getFilters: function (id) {
+            return this._registeredSearches[id].searchFilters;
+        },
+
+        /**
+         * Get the headers for the specified search module
+         *
+         * @param {string} id The search module ID
+         * @return {Array.<string>}
+         */
+        getHeaders: function (id) {
+            return this._registeredSearches[id].searchHeaders;
+        },
+
+        /**
+         * Perform action for item
+         *
+         * @param {string} id Search module to use
+         * @param {string} itemID Search item to handle
+         */
+        handleExecute: function (id, itemID) {
+            var idArray = itemID.split("-"),
+                type = idArray[0],
+                idInt = mathUtil.parseNumber(idArray[1]);
+
+            if (this._registeredSearchTypes[type]) {
+                this._registeredSearchTypes[type].handleExecute(idInt);
+            }
+        },
+
+        /**
+         * Add search module and find initial items, headers, filters
+         *
+         * @param {string} id The search module ID
+         * @param {Array.<string>} types What to search for
+         */
+        registerSearch: function (id, types) {
+            this._registeredSearches[id] = { "searchHeaders": types };
+            this._updateSearchItems(id);
+        },
+
+        /**
+         * Add search type
+         *
+         * @param {object} payload
+            * {Immutable.List.<string>} filters Possible categories for items in this search type
+            * {function} handleExecute Takes item ID (a number) as a parameter. Called when item is confirmed
+            * {function} getOptions Function that returns Immutable.List.<object> of option information 
+                * that can be mapped to a full search options list. List has objects of the form:
+                    * {string} id A number parsed as a string. ID used to perform action if item is selected
```

Comment:
And refer to http://usejsdoc.org/tags-callback.html for how to JSDoc callback functions.